crates/compilers/src/compilers/resolc/compiler.rs

```diff
@@ -149,7 +149,14 @@ impl Resolc {
                     .filter(|x| _resolc_version.is_none_or(|version| version == x.version()))
                     .collect();
 
-                let binary = versions.into_iter().next_back().expect("Can't be empty");
+                let Some(binary) = versions.into_iter().next_back() else {
+                    let message = if let Some(v) = &_resolc_version {
+                        format!("`resolc` v{v} doesn't exist")
```

Comment:
when we set unknown version for solc we get:
Error: Unknown version provided
maybe we should update the error message to `Error: Unknown solc version provided`?
And for resolc
`Error: Unknown `resolc` version provided`
WDYT?

Reply:
> when we set unknown version for solc we get: `Error: Unknown version provided` maybe we should update the error message to `Error: Unknown solc version provided`? And for resolc `Error: Unknown `resolc` version provided` WDYT?

- `Error: unknown version provided` i can't find where this error is produced in code. do you have a link to it? 
- for `solc` we can have both autodetect and specific, so `unsupported `solc` version` can be added as a third message then. WDYT?

Reply:
I do not have it, I just see such message when using unknown solc version

Reply:
> I do not have it, I just see such message when using unknown solc version

i can both come from `svm-rs` and `rvm-rs`